src/metabase/models/dashboard.clj

```diff
@@ -381,10 +381,10 @@
     dashboard))
 
 (def ^:private ParamWithMapping
-  {:name     su/NonBlankStringPlumatic
-   :id       su/NonBlankStringPlumatic
-   :mappings (s/maybe #{dashboard-card/ParamMapping})
-   s/Keyword s/Any})
+  {:id                    su/NonBlankStringPlumatic
```

Comment:
the bug from 24500 is from when calling `dashboard->resolved-params*`, it expects output that has a nonblankstring name.

As described in the PR's description. Name is not required, so we will mark them as optional here.